lib/instrument/transformer/nskeyed.js

```
@@ -0,0 +1,389 @@
+    const klassObj = new klass().decodeArchive(new ArchivedObject(rawObj, this));
+    this.unpackedUids[index.UID] = klassObj;
+    return klassObj;
+  }
+  toObject () {
+    this.unpackArchiveHeader();
+    return this.decodeObject(this.topUID);
+  }
+}
+
+
+/**
+ * Capable of packing an object tree into the NSKeyedArchive format.
+ * Apple's implementation can be found here:
+ * https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSKeyedArchiver.swift
+ */
+class Archive {
+  constructor (inputObject) {
+    this.input = inputObject;
+    this.classMap = {};
+    this.objects = ['$null'];
+  }
+
+  uidForArchiver (archiver) {
+    let val = this.classMap[archiver];
+    if (val) {
+      return new plistlib.UID(val);
+    }
+    this.classMap[archiver] = this.objects.length;
+    val = new plistlib.UID(this.objects.length);
+    this.objects.push({
+      '$classes': [archiver],
+      '$classname': archiver
+    });
+    return val;
+  }
+  archive (obj) {
+    if (typeof obj === 'undefined') {
+      return NULL_UID;
+    }
+    const index = new plistlib.UID(this.objects.length);
+    const objType = typeof obj;
+    if (['number', 'string', 'boolean', 'UID'].indexOf(objType) !== -1) {
+      this.objects.push(obj);
+      return index;
+    }
+    let archiveObj = {};
+    this.objects.push(archiveObj);
+    this.encodeTopLevel(obj, archiveObj);
+    return index;
+  }
+
+  encodeTopLevel (obj, archiveObj) {
+    if (obj instanceof Array) {
+      return this.encodeArray(obj, archiveObj);
+    }
+    if (obj instanceof Set) {
+      return this.encodeSet(obj, archiveObj);
+    }
+    if (obj instanceof Object) {
+      const objName = obj.constructor.name;
+      if (objName in UNARCHIVE_CLASS_MAP) {
+        archiveObj.$class = this.uidForArchiver(objName);
+        obj.encodeArchive(obj, new ArchivingObject(archiveObj, this));
+      } else {
+        return this.encodeDict(obj, archiveObj);
+      }
+    }
+
```

should we throw an exception if none of the above matched?